app/controllers/api/events_controller.rb

```diff
@@ -41,15 +41,15 @@ def handle_notification!
       head :ok
     end
 
-    #
-    # TODO: simplify when notification message format decided.
-    #
     def application_id
+      message.dig('data', 'id')
+    end
+
+    def message
       if request.headers['x-amz-sns-rawdelivery'] == 'true'
```

Comment:
Could wrap this as well `is_aws_sqs_message?` or something not sure someone reading this will necessarily know what the header means.